src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        // Використання умов для вибору відповідного виробника машин на основі типу
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
+            return (List<T>) bulldozerProducer.get();
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+            return (List<T>) excavatorProducer.get();
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> truckProducer = new TruckProducer();
+            return (List<T>) truckProducer.get();
```

Comment:
Similar to lines 15 and 18, this unchecked cast can lead to runtime issues. Consider verifying the type compatibility before casting.